lib/perl/OVH/Bastion/password.inc

```diff
@@ -154,4 +154,24 @@ sub get_hashes_list {
     return R('OK', value => \@ret);
 }
 
+sub is_valid_hash {
+    my %params = @_;
+    my $hash   = $params{'hash'};
+
+    if (not $hash) {
+        return R('ERR_MISSING_PARAMETER', msg => "Missing parameter 'hash'");
+    }
+    elsif ($hash =~ /^\$1\$[a-zA-Z0-9]+\$[a-zA-Z0-9\.\/]+$/) {
+        return R('OK', value => {type => 'md5crypt', hash => $hash});
+    }
+    elsif ($hash =~ /^\$5\$[a-zA-Z0-9]+\$[a-zA-Z0-9\.\/]+$/) {
+        return R('OK', value => {type => 'sha256crypt', hash => $hash});
+    }
+    elsif ($hash =~ /^\$6\$[a-zA-Z0-9]+\$[a-zA-Z0-9\.\/]+$/) {
+        return R('OK', value => {type => 'sha512crypt', hash => $hash});
+    }
```

Comment:
as we've just verified the hash validity, we can now untaint it and pass it up so that our caller can get the untainted version:
```suggestion
    elsif ($hash =~ /^(\$1\$[a-zA-Z0-9]+\$[a-zA-Z0-9\.\/]+)$/) {
        return R('OK', value => {type => 'md5crypt', hash => $1});
    }
    elsif ($hash =~ /^(\$5\$[a-zA-Z0-9]+\$[a-zA-Z0-9\.\/]+)$/) {
        return R('OK', value => {type => 'sha256crypt', hash => $1});
    }
    elsif ($hash =~ /^(\$6\$[a-zA-Z0-9]+\$[a-zA-Z0-9\.\/]+)$/) {
        return R('OK', value => {type => 'sha512crypt', hash => $1});
    }
```

Reply:
Done